lib/plugin-api/hydrogen-kernel.js

```diff
@@ -27,6 +78,70 @@ export default class HydrogenKernel {
     }
   }
 
+  /*
+   * The language of the kernel, as specified in its kernelspec
+   */
+  get language(): string {
+    this._assertNotDestroyed();
+    return this._kernel.language;
+  }
+
+  /*
+   * The display name of the kernel, as specified in its kernelspec
+   */
+  get displayName(): string {
+    this._assertNotDestroyed();
+    return this._kernel.displayName;
+  }
+
+  /*
+   * Add a kernel middleware, which allows intercepting and issuing commands to
+   * the kernel.
```

Comment:
You could add an `@param` docstring here to document that this takes a `Middleware` object. 

We will also need to document somewhere the `Middleware` object api users are supposed to provide. Maybe this deserves it's own gitbook page? I wonder if you have thoughts on this?

Reply:
I added the param docstring.

In terms of actually documenting the API in detail, I'd like to leave that to a future PR. I'm working on some plugins for my own use, and examples would help a lot in terms of explaining what things do.